src/taskgraph/config.py

```diff
@@ -21,94 +17,78 @@
-        Required("workers"): {
-            Required("aliases"): {
-                str: {
-                    Required("provisioner"): optionally_keyed_by("level", str),
-                    Required("implementation"): str,
-                    Required("os"): str,
-                    Required("worker-type"): optionally_keyed_by("level", str),
-                }
-            },
-        },
-        Required("taskgraph"): {
-            Optional(
-                "register",
-                description="Python function to call to register extensions.",
-            ): str,
-            Optional("decision-parameters"): str,
-            Optional(
-                "cached-task-prefix",
-                description="The taskcluster index prefix to use for caching tasks. "
-                "Defaults to `trust-domain`.",
-            ): str,
-            Optional(
-                "cache-pull-requests",
-                description="Should tasks from pull requests populate the cache",
-            ): bool,
-            Optional(
-                "index-path-regexes",
-                description="Regular expressions matching index paths to be summarized.",
-            ): [str],
-            Optional(
-                "run",
-                description="Configuration related to the 'run' transforms.",
-            ): {
-                Optional(
-                    "use-caches",
-                    description="List of caches to enable, or a boolean to "
-                    "enable/disable all of them.",
-                ): Any(bool, list(CACHES.keys())),
-            },
-            Required("repositories"): All(
-                {
-                    str: {
-                        Required("name"): str,
-                        Optional("project-regex"): str,
-                        Optional("ssh-secret-name"): str,
-                        # FIXME
-                        Extra: str,
-                    }
-                },
-                Length(min=1),
-            ),
-        },
-    },
-    extra=ALLOW_EXTRA,
-)
+# TaskPriority type for the priority levels
+TaskPriority = Literal[
+    "highest", "very-high", "high", "medium", "low", "very-low", "lowest"
+]
+
+
+class WorkerAliasSchema(Schema):
+    """Worker alias configuration."""
+
+    provisioner: optionally_keyed_by("level", str)  # type: ignore
+    implementation: str
+    os: str
+    worker_type: optionally_keyed_by("level", str)  # type: ignore
+
+
+class WorkersSchema(Schema, rename=None):
+    """Workers configuration."""
+
+    aliases: Dict[str, WorkerAliasSchema]
+
+
+class Repository(Schema, forbid_unknown_fields=False):
+    """Repository configuration.
+
+    This schema allows extra fields for repository-specific configuration.
+    """
+
+    # Required fields first
+    name: str
+
+    # Optional fields
+    project_regex: Optional[str] = None  # Maps from "project-regex"
+    ssh_secret_name: Optional[str] = None  # Maps from "ssh-secret-name"
```

Comment:
I guess this maps from works because in the base class we pass in the "rename" arg. Out of curiosity, what does it look like if we *wanted* to use underscores rather than dashes?